crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -570,8 +570,28 @@ impl PgManager {
         after: Option<String>,
         last: Option<u64>,
         before: Option<String>,
-        filter: Option<TransactionBlockFilter>,
+        mut filter: Option<TransactionBlockFilter>,
+        self_address: Option<SuiAddress>,
+        // TODO: Do we really need this when filter seems to be able to do the same?
```

Comment:
your comments on the relation stuff makes sense to me, and ig also makes the combination of relation and filter confusing. like, if you set relation = SENT, but then provide sender: anotherAddress, like what does that mean lol. Not blocking but confusing
Similarly on checkpoint, it uses TransactionBlockFilter, which has afterCheckpoint and beforeCheckpoint filters. But this won't make sense on Checkpoint.transactionBlockConnection, presumably you'd want just the transaction blocks that belong to that checkpoint

we can either create validators on these filters or create specific filters for Address, Checkpoint etc. Disad. of the former is that it's confusing to look at the schema and then run into validation checks, disad. of the latter is we clutter up the schema some more.